GenericsHomework/GenericsHomework/Node.cs

```diff
@@ -0,0 +1,141 @@
+﻿using System.Collections;
+using static System.Net.Mime.MediaTypeNames;
+
+namespace GenericsHomework
+{
+    public class Node<T> : ICollection<T>
+    {
+        public T Value { get; }
+
+        public Node<T> Next { get; private set; }
+
+        public Node(T value)
+        {
+            Value = value;
+            Next = this;
+        }
+
+        public Node<T> Append(T value)
+        {
+            if (Exists(value))
+                throw new ArgumentException("Cannot add same value more than once", nameof(value));
+
+            var next = Next;
+            Next = new Node<T>(value)
+            {
+                Next = next
+            };
+            return Next;
+
```

Comment:
while this is correct, it may be easier to read if you break it up a bit more with more specific variable names
```suggestion
       Node<T> newNode = new(value);
       newNode.Next = Next;
       Next = newNode;
       return newNode;
```

Just easier to track what the new next node is vs the currently set next node.